### PR TITLE
Create a publisher for seq

### PR DESCRIPTION
**Pull request by Franklin89**
Looking for suggestions and feedback on this. Also for the message template, I am unsure about a good message that could be displayed.

Usage:

```cs
services.AddHealthChecks()
    .AddSqlServer(configuration.GetConnectionString("DefaultConnection"))
    .AddSeqPublisher(options => 
    {
       options.Endpoint = configuration["Seq:Endpoint"];
       options.ApiKey = configuration["Seq:Apikey"];
     });
```

Dashboards similar to the ones described for App Insights can be created:
image

### CODE REVIEW

**Comment by unaizorrilla** [comment]
Hi  @Franklin89 

Let me this week to check this PR, thanks!

**Comment by CarlosLanderas** [comment]
@Franklin89 could you address the merge problem in this PR? Thanks

**Comment by Franklin89** [comment]
@CarlosLanderas of course, do you have an idea for the message that should be sent to SEQ? 

Currently it just shows this, but I have no Idea at the moment what the best message is for a User: 
image

File: src/HealthChecks.Publisher.Seq/HealthChecks.Publisher.Seq.csproj [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        <TargetFramework>$(NetStandardTargetVersion)</TargetFramework>
+        <PackageLicenseUrl>$(PackageLicenseUrl)</PackageLicenseUrl>
+        <PackageProjectUrl>$(PackageProjectUrl)</PackageProjectUrl>
+        <PackageTags>HealthCheck;Publisher;Seq</PackageTags>
+        <Description>HealthChecks.Publisher.Seq is the health check publisher for Seq.</Description>
+        <Version>$(HealthCheckPublisherPrometheus)</Version>
+        <RepositoryUrl>$(RepositoryUrl)</RepositoryUrl>
+        <Company>$(Company)</Company>
+        <Authors>$(Authors)</Authors>
+        <PackageId>AspNetCore.HealthChecks.Publisher.Seq</PackageId>
```
Comment (unaizorrilla) [inline src/HealthChecks.Publisher.Seq/HealthChecks.Publisher.Seq.csproj]: You need to create a new variable version for this project on  dependencies.props, the version for prometheus and this project can't be the same

File: AspNetCore.Diagnostics.HealthChecks.sln [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.28010.2036
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.28407.52
```
Comment (unaizorrilla) [inline AspNetCore.Diagnostics.HealthChecks.sln]: Are you using VS 2019? this is not a release version, I prefer to mantain 15 version

File: src/HealthChecks.Publisher.Seq/DependencyInjection/SeqHealthCheckBuilderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using System;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class SeqHealthCheckBuilderExtensions
+    {
+        public static IHealthChecksBuilder AddSeqPublisher(this IHealthChecksBuilder builder, Action<SeqOptions> setup)
+        {
+            SeqOptions options = new SeqOptions();
```
Comment (unaizorrilla) [inline src/HealthChecks.Publisher.Seq/DependencyInjection/SeqHealthCheckBuilderExtensions.cs]: use var

File: src/HealthChecks.Publisher.Seq/SeqPublisher.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        private const string METRIC_STATUS_NAME = "AspNetCoreHealthCheckStatus";
+        private const string METRIC_DURATION_NAME = "AspNetCoreHealthCheckDuration";
+
+        private readonly SeqOptions _options;
+        private readonly HttpClient _httpClient;
+
+        public SeqPublisher(SeqOptions options)
+        {
+            _options = options;
+            _httpClient = new HttpClient
```
Comment (unaizorrilla) [inline src/HealthChecks.Publisher.Seq/SeqPublisher.cs]: We are changing the code to use IHttpClientFactory, can you check the Uri HealthCheck or IdSvr in order  to check this?

File: src/HealthChecks.Publisher.Seq/SeqPublisher.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            {
+                BaseAddress = new Uri(options.Endpoint)
+            };
+        }
+
+        public void Dispose()
+        {
+            if (_httpClient != null)
+            {
+                _httpClient.Dispose();
```
Comment (unaizorrilla) [inline src/HealthChecks.Publisher.Seq/SeqPublisher.cs]: If you use IhttpClientFactory you don't need to dispose this

**Comment by unaizorrilla** [comment]
Hi @Franklin89

can you fix this minor issues?

**Comment by Franklin89** [comment]
@unaizorrilla I made the requested changes. Let me know if there is anything else that should be done.

**Comment by unaizorrilla** [comment]
Merged with some minor changes! Matteo, can you try to add a Seq docker image in the docker-compose in order to test this publisher?

**Pull request merged**